src/environment-logs-provider.ts

```diff
@@ -0,0 +1,168 @@
+import * as vscode from "vscode";
+import { Environment } from "./env0-environments-provider";
+import { apiClient } from "./api-client";
+import stripAnsi from "strip-ansi";
+import { DeploymentStatus, DeploymentStepStatus } from "./types";
+import axios from "axios";
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+const pollStepLogsInterval = 1000;
+
+export class EnvironmentLogsProvider {
+  private static environmentLogsOutputChannel: vscode.OutputChannel;
+  private readonly abortController = new AbortController();
+  private _isAborted = false;
+  private readonly stepsAlreadyLogged: string[] = [];
+  constructor(private readonly env: Environment) {
+    this.logDeployment(this.env.latestDeploymentLogId).catch((e) => {
+      if (!axios.isCancel(e)) {
+        throw e;
+      }
+    });
+  }
+
+  abort() {
+    this.abortController.abort();
+    this._isAborted = true;
+    EnvironmentLogsProvider.environmentLogsOutputChannel.clear();
+  }
+
+  get isAborted() {
+    return this._isAborted;
+  }
+
+  private log(value: string) {
+    if (!this.isAborted) {
+      const channel = EnvironmentLogsProvider.environmentLogsOutputChannel;
+      if (!channel) {
+        throw new Error(
+          "environment logs output channel used before initialized"
+        );
+      }
+      channel.appendLine(value);
+    }
+  }
+
+  private async logDeployment(deploymentId: string) {
+    let previousStatus;
+    EnvironmentLogsProvider.environmentLogsOutputChannel.show();
+
+    const deployment = await apiClient.getDeployment(
+      deploymentId,
+      this.abortController
+    );
+
+    if (deployment.status === DeploymentStatus.QUEUED) {
+      this.log(`Deployment is queued! Waiting for it to start...`);
+    }
```

Comment:
update `previousStatus`, you already queried the deployment, no need for another logging within the while if it's still `QUEUED`

Reply:
you didn't store the status, it won't reach the `previous === QUEUED`, no?

Reply:
i am not sure i understand,
but i did assign the status value to the previousStatus variable in the end of the loop [here](https://github.com/env0/vscode-env0/pull/40/files#diff-55ec115f8bce8531aada11dab10df383176a314d2a8491ee5cac38f2c6da6d81R95)

Reply:
you store _only_ at the end of the loop. you put a `continue` in the condition, so the value for `previousStatus` isn't updated

Reply:
Good catch, 16fbf35f896c503d459340da44d8f23f7a3e4860